### PR TITLE
Add Kubernetes backend

### PR DESCRIPTION
**Pull request by aiordache**
Add Kubernetes context type and backend.

Relates to  #1145 #1146 #1147

```
$ docker context create kubernetes kubectx
Successfully created kubernetes context "kubectx"

$ docker context use kubectx
kubectx
```
```
$ kubectl get svc
NAME         TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   23h
```

```
$ cat docker-compose.yml 
version: '3.7'
services:
  db:
    image: mysql:8.0.19
    command: '--default-authentication-plugin=mysql_native_password'
    restart: always
    ports:
      - 3306:3306
    volumes:
      - db_data:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=somewordpress
      - MYSQL_DATABASE=wordpress
      - MYSQL_USER=wordpress
      - MYSQL_PASSWORD=wordpress
  wordpress:
    image: wordpress:latest
    ports:
      - 80:80
    restart: always
    environment:
      - WORDPRESS_DB_HOST=db
      - WORDPRESS_DB_USER=wordpress
      - WORDPRESS_DB_PASSWORD=wordpress
      - WORDPRESS_DB_NAME=wordpress
volumes:
  db_data:

```
```
$ docker compose up
2021/01/19 15:10:57 creating 5 resource(s)
2021/01/19 15:10:57 Release status:  deployed
2021/01/19 15:10:57 Install complete
```
```
$ docker compose ls
NAME                STATUS
test                deployed
```
```
$ kubectl get svc
NAME         TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)          AGE
db           NodePort    10.103.138.152   <none>        3306:32116/TCP   3s
kubernetes   ClusterIP   10.96.0.1        <none>        443/TCP          23h
wordpress    NodePort    10.110.94.205    <none>        80:30337/TCP     3s
```

```
$ docker compose down
2021/01/19 15:11:02 uninstall: Deleting test
2021/01/19 15:11:02 Starting delete for "wordpress" Service
2021/01/19 15:11:02 Starting delete for "db" Service
2021/01/19 15:11:02 Starting delete for "db" Deployment
2021/01/19 15:11:02 Starting delete for "wordpress" Deployment
2021/01/19 15:11:02 Starting delete for "db-data" PersistentVolumeClaim
2021/01/19 15:11:02 purge requested for test
2021/01/19 15:11:02 Uninstallation complete
```

### CODE REVIEW

File: api/context/store/store.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ const (
 	// LocalContextType is the endpoint key in the context endpoints for a new
 	// local backend
 	LocalContextType = "local"
+	// KubeContextType is the endpoint key in the context endpoints for a new
+	// kube backend
+	KubeContextType = "kube"
```
Comment (gtardif) [inline api/context/store/store.go:60]: Nit: "kubernetes" ?

**Review by gtardif — COMMENTED** [review]
A few minor things... will need to play with the runtime a bit more later on

**Review by ndeloof — APPROVED** [review]
I wish the helm-prorotype git history could have been preserved.
LGTM anyway

File: kube/utils/errors.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+*/
+
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+func CombineErrors(errors []error) error {
```
Comment (gtardif) [inline kube/utils/errors.go:26]: We're using "github.com/hashicorp/go-multierror" in many places, looks like we could use it instead of CombineErrors

**Review by gtardif — APPROVED** [review]
Some minor things (dead code & multi error), and other comments more for follow up PRs, LGTM

**Comment by ndeloof** [comment]
the charts.API seems to be an unecessary abstraction while we already have compose.API to define the expected interface. This is why I first created https://github.com/docker/helm-prototype/issues/10 so that we adapt help-prototype code to match the compose-cli API and we don't get such an API mismatch

**Pull request merged**